File: LeetCode/Easy/1929 ConcatenationOfArray/1929ConcatenationOfArray.py
from typing import List

class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        return nums + nums

if __name__ == "__main__":
    sol = Solution()
    nums = [1, 2, 1]
    ans = sol.getConcatenation(nums)
    print(ans)

File: LeetCode/Easy/1929 ConcatenationOfArray/1929ConcatenationOfArray.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(2 * n);
        
        for (int i = 0; i < n; i++) {
            ans[i] = nums[i];
            ans[i + n] = nums[i];
        }
        
        return ans;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {1, 2, 1};
    vector<int> ans = sol.getConcatenation(nums);

    cout << "[";
    for (size_t i = 0; i < ans.size(); ++i) {
        cout << ans[i] << (i == ans.size() - 1 ? "" : ", ");
    }
    cout << "]" << endl;

    return 0;
}